### PR TITLE
ETL into pdap/data-intake

### PR DESCRIPTION
**Pull request by EricTurner3**
ETL Package refactored to read from `schema.json` file and will now:
- search for `agency_id` and grab the record
- search for all available `dataset_id`, insert if they do not exist
- search for `data_type`, verify it exists, then grab the same named table in `pdap/data-intake`
- sync the `mapping` in `schema.json` to match the columns for the `pdap/data-intake` table
- output all the modifications (sync) back out to the `schema.json` file

Still work-in-progress:
- use the synced `mapping` from the database to actually enumerate the scraped data records and ingest them into `pdap/data-intake`

### CODE REVIEW

**Comment by EricTurner3** [comment]
Successfully loaded data into DoltHub!

- [pdap/data-intake](https://www.dolthub.com/repositories/pdap/data-intake/pulls/7) (all the data in the bulletins tables are from this etl process)
- [pdap/datasets](https://www.dolthub.com/repositories/pdap/datasets/pulls/39/compare) (two new datasets for the agency)

File: USA/CA/butte_county/college/chico/data/daily_bulletins/CSUCPD_2021_05_19_daily_bulletin [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+6,Cntl Sub Paraphernalia,21-0223,5/11/2021 09:14,5/11/2021  09:14,"Walnut St/W 1ST St & Nord Av, Chico",Crime Cleared by Arrest 5/11/2021
+7,Indecent Exposure     Annoy/Etc Child -18,21-0222,5/10/2021 16:11,5/10/2021  15:14,Bike Path,Closed 5/11/2021
+8,Cntl Sub Paraphernalia     Poss Controlled Substance-misd,21-0221,5/10/2021 15:51,5/10/2021  15:51,Wildcat Wellness Center,Crime Cleared by Arrest 5/10/2021
+9,Outside Warrant Arrest MISD,21-0220,5/10/2021 11:11,5/10/2021  11:11,0 Block of Main,Crime Cleared by Arrest 5/10/2021
+10,Outside Warrant Arrest MISD     Outside Warrant Arrest MISD,21-0219,5/10/2021 10:53,5/10/2021  10:53,0 Block of Main,Crime Cleared by Arrest 5/10/2021
+11,Mfg/import/sale/poss of metal knuckles,21-0217,5/9/2021 01:29,5/9/2021  01:29 - 5/9/2021 02:43,Tonys Food & Liquor,Crime Cleared by Arrest 5/9/2021
+12,Poss Controlled Substance-misd,21-0218,5/9/2021 01:29,5/9/2021  01:29 - 5/9/2021 02:43,Tonys Food & Liquor,Crime Cleared by Arrest 5/9/2021
+13,Outside Warrant Arrest FELONY     Cntl Sub Paraphernalia,21-0216,5/8/2021 21:45,5/8/2021  21:45 - 5/8/2021 23:02,"Mansion Av/Arcadian Av, Chico",Crime Cleared by Arrest 5/8/2021
```
Comment (CaptainStabs) [inline USA/CA/butte_county/college/chico/data/daily_bulletins/CSUCPD_2021_05_19_daily_bulletin:15]: May want to truncate the data.

File: USA/CA/butte_county/college/chico/etl.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+'''
+    schema_load:
+        - schema file JSON
+        - branch name (override)
+'''
+schema = main.schema_load(json.load(open('schema.json', 'r')))
+
+# overwrite the schema.json with the changes from the etl
+with open('schema.json', 'w+') as outfile:
+    json.dump(schema, outfile, indent=4)
```
Comment (CaptainStabs) [inline USA/CA/butte_county/college/chico/etl.py:20]: Newline missing

Comment (CaptainStabs) [inline USA/CA/butte_county/college/chico/etl.py:20]: Add newline at end of file

File: USA/CA/butte_county/college/chico/main.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+# run the scrapers for this agency
+
+print('Running Chico - Daily Bulletins Scraper')
+import chico_daily
+print('Running Chico - Crimegraphics Bulletins Scraper')
+import crimegraphics_bulletin
+
+# then run the etl process
+print('Running Chico - ETL')
+import etl
```
Comment (CaptainStabs) [inline USA/CA/butte_county/college/chico/main.py:11]: Add newline

File: USA/CA/butte_county/college/chico/schema.json [hunk 4]
```diff
@@ -1,25 +1,63 @@
...
+                "ActivityPlace": "ActivityPlace",
+                "ActivityStreet": "ActivityStreet",
+                "ActivityCity": "ActivityCity",
+                "datasets_id": "__dataset_id__",
+                "date_insert": "__date__"
+            }
         }
     ]
-    
 }
```
Comment (CaptainStabs) [inline USA/CA/butte_county/college/chico/schema.json:63]: add newline

File: common/etl/main.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+            pdap_dolt.load_csv_data_from_schema_map(intake, intake_db_cols, intake_table_name, dataset, datafile)
+
+
+    # 5 - Commit the Changes to Dolt
+    # For Now I like to manually review the changes and commit
+    #pdap_dolt.commit(dolt, intake)
+    print('Review the changes with dolt and commit them!')
+
+    # 6- Write out Schema.json!
+    return schema
```
Comment (CaptainStabs) [inline common/etl/main.py:56]: add newline

File: common/etl/pdap_dolt.py [hunk 6]
```diff
@@ -271,17 +389,170 @@ def load_data(intake, dataset_record, file):
...
+        dolt.remote(add=True, name='origin', url=dolthub_fullrepo)
+        dolt.commit('Data Added from ETL Script')
         dolt.push('origin', branch, set_upstream='origin')
+
+        intake.remote(add=True, name='origin', url=intake_fullrepo)
+        intake.commit('Data Added from ETL Script')
+        intake.push('origin', intake_branch, set_upstream='origin')
         print('  [*] Done!')
     else:
         print(  '[!] ERROR: Cannot push to main branch. Aborting. Use dolt cli to migrate and finalize commit')
```
Comment (CaptainStabs) [inline common/etl/pdap_dolt.py:558]: Add newline

**Review by CaptainStabs — CHANGES_REQUESTED** [review]
Add the newline to the commented files and it should be good to go. 

https://gcc.gnu.org/legacy-ml/gcc/2003-11/msg01568.html

**Comment by EricTurner3** [comment]
I updated my settings to Insert NewLines and will do PEP8 formatting and unit tests in the next PR. I just do not want to increase this PR as it functionally is working fine and our pipeline does not currently have specific formatting requirements.

**Review by CaptainStabs — APPROVED** [review]
Other than the formatting (which he will fix in a different PR), the code looks good.

**Pull request merged**